crates/paimon/src/spec/schema.rs

```diff
@@ -51,3 +51,176 @@ pub struct DataField {
     typ: DataType,
     description: Option<String>,
 }
+
+impl DataField {
+    pub fn new(id: i32, name: String, typ: DataType, description: Option<String>) -> Self {
+        Self {
+            id,
+            name,
+            typ,
+            description,
+        }
+    }
+
+    pub fn id(&self) -> i32 {
+        self.id
+    }
+
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    pub fn data_type(&self) -> &DataType {
+        &self.typ
+    }
+
+    pub fn description(&self) -> Option<&String> {
+        self.description.as_ref()
+    }
+
+    pub fn new_id(&self, new_id: i32) -> Self {
+        Self::new(
+            new_id,
+            self.name.clone(),
+            self.typ.clone(),
+            self.description.clone(),
+        )
+    }
+
+    pub fn new_name(&self, new_name: String) -> Self {
+        Self::new(
+            self.id,
+            new_name,
+            self.typ.clone(),
+            self.description.clone(),
+        )
+    }
+
+    pub fn new_description(&self, new_description: String) -> Self {
+        Self::new(
+            self.id,
+            self.name.clone(),
+            self.typ.clone(),
+            Some(new_description),
+        )
+    }
+
+    pub fn as_sql_string(&self) -> String {
+        self.format_string(&self.typ.to_string())
+    }
+
+    fn format_string(&self, type_string: &str) -> String {
+        match &self.description {
+            Some(desc) => format!(
+                "{} {} '{}'",
+                self.escape_identifier(&self.name),
+                type_string,
+                self.escape_single_quotes(desc)
+            ),
+            None => format!("{} {}", self.escape_identifier(&self.name), type_string),
+        }
+    }
+
+    fn escape_identifier(&self, identifier: &str) -> String {
```

Comment:
It's a bit odd to have those functions as methods of `DataField`. Perhaps we should move them out.

Reply:
You are right, I put these functions outside as utility functions